src/store/epics/wallets.ts

```diff
@@ -54,18 +58,25 @@ export const unlockWallet$: RootEpic = (action$, state$, { decryptPrivateKey })
     filter(isActionOf(unlockWalletRequest)),
     withLatestFrom(state$),
     map(([_, state]) => {
+      const now = Date.now()
+      if (failureAttemptHandler(state.wallets.failureAttemptTimestamps, now).afterLockInTime) {
```

Comment:
I think the `failureAttemptHandler` should be used as a dependency, not directly imported into this module. It makes it easier to test and isolate, and is the recommended practice, as well as being the practice we're already following with the epics. If you look at the `root-epic.ts` file for examples of how to include dependencies you'll see they become available as parameters to the epic

Reply:
I have refactored the code and removed the above mentioned part.